side bar styling & design update & hooks folder

Akpjunior94 opened the pull request:
**This pull request makes the following changes**
* Cleans up #81 PR based on review from @Ifycode 


**Testing checklist**

- [x] The design is responsive and implemented as stated in the issue description.
- [x] Accessibility is left/implemented as stated in the issue description.
- [x] Content area [right] looks and functionality is implemented as stated in the issue description
- [x] Sidebar [left] looks and functionality is implemented as stated in the issue description
- [x]  No external library is used
- [x] I certify that I ran my checklist
- [x] I certify that I ran my checklist

Ping @code-collabo/web-app

AbrahamAkp committed:
side bar styling & design update & hooks folder

Akpjunior94 commented:
@Ifycode Here is the updated PR that fixes reviews on this PR #81 I hope I am able to follow your thought process that has been put in place 😆

Ifycode commented:
This is really clean @Akpjunior94 
Love what I see from checking briefly.
Thorough review coming later today.

Ifycode commented on client/src/apps/code-collabo/styles/app.main.js:
```diff
@@ -22,6 +22,24 @@ export default css.global`
+    .app__menubar__nav__items {
+        gap: 10px;
+        width: 100%;
+        padding: 0 20px;
```
**Mobile:**

Please **use** width here (instead of padding for this `.app__menubar__nav__items` container item. As I have said we don't want to use padding on the left and right of container items).

Explanation:
On the figma design the sidebar width for mobile is 300, and then the width of the button(s) is 240. The space at the sides is 60 (i.e. 30 on the left and 30 on the right).

In the code, you can use the idea to get the percentage of the `.app__menubar__nav__items` container for mobile - use that as width (you can just use the CSS calc() function for this).
Then use the auto margin on the `.app__menubar__nav__items` container to centralize it - that way you don't have to bother about what the sides will be as they are just set dynamically. Saying this here also because that gap spacing is incorrect for mobile based on "Explanation" above.

Akpjunior94 replied:
updated

Ifycode commented on client/src/apps/code-collabo/styles/app.main.js:
```diff
@@ -68,6 +86,19 @@ export default css.global`
         }
 
+        .app__logo {
+            margin-bottom: 82px;
```
Margin bottom is correct 👍🏼
According to the figma design when it comes to the app logo (Desktop):
There's supposed to be a margin-top of 12 here too.

Akpjunior94 replied:
Updated

Ifycode commented on client/src/apps/code-collabo/styles/app.main.js:
```diff
@@ -68,6 +86,19 @@ export default css.global`
+        }
+
+        .app__menubar__nav__link {
+            padding: 15px 0px;
```
Desktop Padding top and bottom - 16, not 15.

Akpjunior94 replied:
@Ifycode Done

Ifycode commented on client/src/apps/code-collabo/styles/app.main.js:
```diff
@@ -68,6 +86,19 @@ export default css.global`
+        }
+
+        .app__menubar__nav__items {
+            gap: 20px
```
**Desktop:**

Please **add** width here (instead of padding for this `.app__menubar__nav__items` container item. As I have said we don't want to use padding on the left and right of container items).

Explanation:
On the figma design the sidebar width for **desktop** is 360, and then the width of the button(s) is 300. The space at the side is 60 (i.e. 30 on the left and 30 on the right).

In the code, you can use the idea to get the percentage of the `.app__menubar__nav__items` container for desktop - use that as width (you can just use the CSS calc() function for this).
The use the auto margin on the `.app__menubar__nav__items` container to centralize it - that way you don't have to bother about what the sides will be as they are just set dynamically. Saying this here also because that gap spacing is incorrect for desktop based on "Explanation" above.

Akpjunior94 replied:
@Ifycode Done

Ifycode commented on client/src/apps/code-collabo/styles/app.main.js:
```diff
@@ -22,6 +22,24 @@ export default css.global`
     }
 
+    .app__logo {
+        margin-bottom: 40px;
```
Mobile Margin bottom - 50, not 40 according to figma design.

Ifycode commented on client/src/apps/code-collabo/components/AppPageStructure.tsx:
```diff
@@ -36,13 +36,27 @@ export default function PageStructure({ children }: { children: ReactNode }) {
+          </Link>
+          <Link className='app__menubar__nav__link lib__flex-center' href='/code-collabo/projects'>
+            <Image src='/code-collabo/project-icon.png' alt='project-icon' width='17' height='15' />
+            Projects
```
Projects link icon: Something changes the color of the icon to white or so on the page - so the icon is not visible, help look into this.

Akpjunior94 replied:
it is whie that is why it is not visible. it is the Active state that is why. The designer will have to provide an inactive image icon for it since we are using the exported image icons for now.

Ifycode replied:
Oh... Ok, nice. Thank you. I totally forgot. Thank you. I've add it as a reminder comment for @Geefundesign on figma: https://www.figma.com/file/dyR55ZfOIQBvIJo6UMxZC9?node-id=47:13546&mode=design#731401524

You can remind him too (on figma - since that is his workspace, or on discord public channel).

Ifycode commented on client/src/apps/code-collabo/styles/app.main.js:
```diff
@@ -22,6 +22,24 @@ export default css.global`
+        border-radius: 10px;
+    }
+
+    .app__menubar__nav__items {
```
Both Mobile and Desktop:
Let `.app__menubar__nav__items` styles class come before `.app__menubar__nav__link` since that is the container.

Akpjunior94 replied:
@Ifycode Done

Ifycode commented:
@AbrahamAkp Write a javasript/react code that toggles things on and off. To show that it works, you can just show and hide the sidebar (on Mobile only) with this toggle.

AbrahamAkp committed:
Review changes updates & toggle functionality

Akpjunior94 commented:
> @AbrahamAkp Write a javasript/react code that toggles things on and off. To show that it works, you can just show and hide the sidebar (on Mobile only) with this toggle.

Hi @Ifycode I have implemented the toggle functionality. Please have a look and I hope it follows the rules. I also changed some logic like the <=600 we are using for isMobile i changed it to <600 instead.

Ifycode commented on client/src/apps/code-collabo/hooks/useScreenDimensions.ts:
```diff
@@ -16,7 +16,7 @@ const useScreenDimensions = (): ScreenDimensions => {
       const newScreenWidth = window.innerWidth;
       setScreenDimensions({
-        isMobile: newScreenWidth <= 600,
+        isMobile: newScreenWidth < 600,
```
Leave it as `<=` so that it ryhmes with the css.
I will leave you more reviews for others on or before midnight.

Akpjunior94 replied:
@Ifycode Done.

Ifycode commented on client/src/apps/code-collabo/components/AppPageStructure.tsx:
```diff
@@ -20,6 +20,11 @@ export default function PageStructure({ children }: { children: ReactNode }) {
+
+  const toggleSidebar = () => {
+    setIsSidebarOpen(!isSidebarOpen);
+  };
```
Can we make this logic an external function/hook? We'll be needing it in other places on the website.

Akpjunior94 replied:
Done

Ifycode replied:
Help with the todo for filters on mobile: https://github.com/code-collabo/collabo-community-app/pull/86/files#diff-fb1c10e8b38585af4f358b5c6a960930ec17b53c2d42a7c43d5f93bb7f8b4216L65
And toggle accordingly.

Ifycode replied:
Help with the todo for filters on mobile: https://github.com/code-collabo/collabo-community-app/pull/86/files#diff-fb1c10e8b38585af4f358b5c6a960930ec17b53c2d42a7c43d5f93bb7f8b4216L65
And toggle accordingly.

> There is a filter component already, you don't have to remake another one

AbrahamAkp committed:
toogle hook & revert screenDimension size

AbrahamAkp committed:
Mobile filter toggle

Ifycode committed:
Merge pull request #88 from code-collabo/mobile-filter-sidebar

Mobile filter toggle

Ifycode commented:
Merging temporarily for further cleanup fixes to happen alongside the sidebar. It's easy to see the 2 together.
Plus, we can have a new PR with less comments.
